test/AspNet.Security.OAuth.Providers.Tests/AmoCrm/AmoCrmTests.cs

```diff
@@ -0,0 +1,44 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace AspNet.Security.OAuth.AmoCrm
+{
+    public class AmoCrmTests : OAuthTests<AmoCrmAuthenticationOptions>
+    {
+        public override string DefaultScheme => AmoCrmAuthenticationDefaults.AuthenticationScheme;
+
+        protected internal override void RegisterAuthentication(AuthenticationBuilder builder)
+        {
+            builder.AddAmoCrm(options =>
+            {
+                options.Account = "example";
+                ConfigureDefaults(builder, options);
+            });
+        }
+
+        [Theory]
+        [InlineData(ClaimTypes.NameIdentifier, "100500")]
+        [InlineData(ClaimTypes.Name, "John")]
+        [InlineData(ClaimTypes.Email, "john@john-smith.local")]
+        public async Task Can_Sign_In_Using_AmoCrm(string claimType, string claimValue)
```

Comment:
Please add a test case for `ClaimTypes.Surname`.

Reply:
@martincostello, done!